zeroex/orderwatch/order_watcher.go

```diff
@@ -1158,89 +1158,53 @@ func (w *Watcher) convertValidationResultsIntoOrderEvents(
 			}
 			orderEvents = append(orderEvents, orderEvent)
 		} else {
-			// The order is expired if ExpirationTimeSeconds is equal to or less than the timestamp
+			// The timestamp is valid if its expiration time is greater than the timestamp
 			// of the validation block.
-			validationBlockTimeStampSeconds := big.NewInt(validationBlock.Timestamp.Unix())
-			isOrderExpired := order.ExpirationTimeSeconds.Cmp(validationBlockTimeStampSeconds) != 1
+			validationBlockTimestampSeconds := big.NewInt(validationBlock.Timestamp.Unix())
+			timestampIsValid := order.ExpirationTimeSeconds.Cmp(validationBlockTimestampSeconds) == 1
+			isOrderUnexpired := order.IsRemoved && timeStampIsValid
 
-			if oldFillableAmount.Cmp(newFillableAmount) == 0 {
-				// If order was previously expired, check if it has become unexpired
-				if order.IsRemoved && oldFillableAmount.Cmp(big.NewInt(0)) != 0 && !isOrderExpired {
-					w.rewatchOrder(order, nil, validationBlock)
-					orderEvent := &zeroex.OrderEvent{
-						Timestamp:                validationBlock.Timestamp,
-						OrderHash:                order.Hash,
-						SignedOrder:              order.SignedOrder(),
-						FillableTakerAssetAmount: order.FillableTakerAssetAmount,
-						EndState:                 zeroex.ESOrderUnexpired,
-					}
-					orderEvents = append(orderEvents, orderEvent)
-				}
-				// No important state-change happened. Still want to update lastValidatedBlock
-				w.updateOrderLastValidatedBlock(order, validationBlock)
-				continue
-			}
-			if oldFillableAmount.Cmp(big.NewInt(0)) == 1 && oldAmountIsMoreThenNewAmount {
-				// If order was previously expired, check if it has become unexpired
-				if order.IsRemoved && oldFillableAmount.Cmp(big.NewInt(0)) != 0 && !isOrderExpired {
-					w.rewatchOrder(order, newFillableAmount, validationBlock)
-					orderEvent := &zeroex.OrderEvent{
-						Timestamp:                validationBlock.Timestamp,
-						OrderHash:                order.Hash,
-						SignedOrder:              order.SignedOrder(),
-						FillableTakerAssetAmount: order.FillableTakerAssetAmount,
-						EndState:                 zeroex.ESOrderUnexpired,
-					}
-					orderEvents = append(orderEvents, orderEvent)
-				} else {
-					w.updateOrderFillableTakerAssetAmount(order, newFillableAmount, validationBlock)
-				}
-				// Order was filled, emit event
+			if isOrderUnexpired {
+				w.rewatchOrder(order, newFillableAmount, validationBlock)
 				orderEvent := &zeroex.OrderEvent{
 					Timestamp:                validationBlock.Timestamp,
-					OrderHash:                acceptedOrderInfo.OrderHash,
+					OrderHash:                order.Hash,
 					SignedOrder:              order.SignedOrder(),
-					EndState:                 zeroex.ESOrderFilled,
-					FillableTakerAssetAmount: acceptedOrderInfo.FillableTakerAssetAmount,
-					ContractEvents:           orderHashToEvents[order.Hash],
+					FillableTakerAssetAmount: order.FillableTakerAssetAmount,
```

Comment:
`order.FillableTakerAssetAmount` is the old fillable amount, which makes this a behavioral change. This change does have some advantages, namely that it makes reconstructing the order book state from order events a little bit more intuitive. But the downside is that it is slightly less efficient because MMs and other users would have to parse the next order event in order to see the latest fillable amount (even though  we already know what that is).

I think the two approaches are fairly equal in terms of pros/cons, but that means IMO we should keep the behavior the same instead of changing it. Was this just an oversight or was it an intentional change on your part? Curious to hear what you think?

```suggestion
					FillableTakerAssetAmount: newFillableAmount,
```